tests/cupy_tests/core_tests/test_internal.py

```diff
@@ -49,9 +50,27 @@ def test_tuple(self):
     def test_int(self):
         assert internal.get_size(1) == (1,)
 
+    def test_numpy_int(self):
+        assert internal.get_size(numpy.int32(1)) == (1,)
+
+    def test_numpy_zero_dim_ndarray(self):
+        assert internal.get_size(numpy.array(1)) == (1,)
+
+    def test_tuple_of_numpy_scalars(self):
+        assert internal.get_size((numpy.int32(1), numpy.array(1))) == (1, 1)
+
+    @pytest.mark.parametrize(
+        'value', [True, numpy.bool_(True), numpy.array(True, dtype='?')])
+    def test_bool(self, value):
+        with pytest.raises(TypeError):
+            internal.get_size(value)
+        with pytest.raises(TypeError):
+            internal.get_size((value, value))
+
     def test_float(self):
-        with pytest.raises(ValueError):
-            internal.get_size(1.0)
+        # `internal.get_size` is not responsible to interpret values as
+        # integers.
+        assert internal.get_size(1.0) == (1.0,)
```

Comment:
If `internal.get_size` is expected not to raise errors, we should add a test checks if `cupy.ndarray(shape=[1.0, 2.0])` raises TypeError?

Reply:
Sure!